dapps/marketplace/src/pages/transaction/_ReviewAndFinalization.js

```diff
@@ -0,0 +1,200 @@
+import React, { Component, Fragment } from 'react'
+import { fbt } from 'fbt-runtime'
+
+import StarRating from 'components/StarRating'
+import Stages from 'components/TransactionStages'
+import AddData from 'pages/transaction/mutations/AddData'
+import Link from 'components/Link'
+
+const RATED_OFFERS_KEY = 'rated_offers'
+class ReviewAndFinalization extends Component {
+  state = {
+    rating: 0,
+    review: '',
+    offerRated: false
+  }
+
+  componentDidMount() {
+    const offersRatedString = localStorage.getItem(RATED_OFFERS_KEY)
+    const offer = this.props.offer
+
+    if (offersRatedString) {
+      this.setState({
+        offerRated: JSON.parse(offersRatedString).includes(offer.id)
+      })
+    }
+  }
+
+  renderFinalization({ loading, offer, isSeller, isBuyer }) {
+    return (
+      <div className={`transaction-progress${loading ? ' loading' : ''}`}>
+        <div className="top">
+          <h4>
+            {isSeller && (
+              <fbt desc="Progress.yourSaleComplete">Your sale is complete.</fbt>
+            )}
+            {isBuyer && (
+              <fbt desc="Progress.yourPurchaseComplete">
+                Your purchase is complete.
+              </fbt>
+            )}
+          </h4>
+          <Stages className="mt-4" mini="true" offer={offer} />
+          {isBuyer && (
+            <Fragment>
+              <div className="help mb-0 mt-4">
+                <fbt desc="Progress.seeOtherListings">
+                  See what other listings are available on Origin.
+                </fbt>
+              </div>
+              <div className="actions">
+                <div className="btn btn-link">
+                  <Link to="/">
+                    <fbt desc="Progress.viewListings">View Listings</fbt>
+                  </Link>
+                </div>
+              </div>
+            </Fragment>
+          )}
+          {isSeller && (
+            <div className="actions">
+              <div className="btn btn-link">
+                <Link to="/create">
+                  <fbt desc="Progress.createAnotherListing">
+                    Create Another Listing
+                  </fbt>
+                </Link>
+              </div>
+            </div>
+          )}
+        </div>
+      </div>
+    )
+  }
+
+  rateOffer(offerId) {
+    const offersRated = JSON.parse(
+      localStorage.getItem(RATED_OFFERS_KEY) || '[]'
+    )
+    offersRated.push(offerId)
+    this.setState({
+      offerRated: true
+    })
+    localStorage.setItem(RATED_OFFERS_KEY, JSON.stringify(offersRated))
+  }
+
+  renderRating({ offer, loading, isSeller, isBuyer }) {
```

Comment:
I think we should try to use functional components instead of render methods going forward so we can try to transition away from React classes

Reply:
Cool converting to functional component with `useState` and `useEffect`